### PR TITLE
Fix tests so that they work with dev tibble

### LINKED ISSUE #3918: dev tibble breaks dplyr unit tests
See the `dev-tibble` branch.

### PR DESCRIPTION
**Pull request by krlmlr**
and also tibble 1.4.2.

Closes #3918.

### CODE REVIEW

File: R/group_data.R [hunk 1]
```diff
@@ -33,13 +33,13 @@ group_data <- function(.data) {
 #' @export
 group_data.data.frame <- function(.data) {
   rows <- list(seq_len(nrow(.data)))
-  tibble(.rows=rows)
+  tibble(.rows := rows)
```
Comment (romainfrancois) [inline R/group_data.R]: Is it worth mentioning that in `?tibble` ?

File: tests/testthat/test-joins.r [hunk 2]
```diff
@@ -1024,7 +1024,7 @@ test_that("joins reject data frames with duplicate columns (#3243)", {
 
   expect_error(
     left_join(df1, df2, by = c("x", "y")),
-    "Column `x` must have a unique name",
+    "name",
```
Comment (romainfrancois) [inline tests/testthat/test-joins.r]: That looks too "vague"

**Comment by romainfrancois** [comment]
Can you also un skip the test that says "until updating tibble" please.

File: R/group_data.R (hunk 1 above)
Comment (krlmlr) [reply R/group_data.R]: Done now.

File: tests/testthat/test-joins.r (hunk 2 above)
Comment (krlmlr) [reply tests/testthat/test-joins.r]: The error messages come from _tibble_, can't be vague enough here.

**Pull request merged**